src/pythonjsonlogger/core.py

```diff
@@ -117,7 +118,8 @@ def merge_record_extra(
     """
     if rename_fields is None:
         rename_fields = {}
-    for key, value in record.__dict__.items():
+    record_dict = record.__dict__ if not defaults else (defaults | record.__dict__)
```

Comment:
note: `dict | dict` is not compatible with python 3.8, we'll need to use an older syntax.

I think it would make sense to remove the `not` part of this.
```suggestion
    # note: merged = d1 | d2 not available until py39
    record_dict = {**defaults, **record.__dict__} if defaults else record.__dict__
```

Reply:
Actually re-reading the code (because github doesn't show the full context by default), I'm wondering if this should be moved to [`Base.format`](https://github.com/nhairs/python-json-logger/pull/26/files#diff-ccc072477bf6bb0c98704e583b0836f99660281b0441375326e18d0ad474c6b9R256) specifically:

```
# before
log_record: LogRecord = {}
# after
log_record: LogRecord = self.defaults.copy() if self.defaults else {}

Reply:
So originally I wished for it to be close to the "extras" parameter processing, as it made much sense. I may have overlooked your solution though and I think it's a good idea.